### PR TITLE
Migrate from JSCS to ESLint

### PR DESCRIPTION
**Pull request by bogdan-stefan-hs**
## Need

``` gherkin
As a developer
I want to have a good linter enabled for my project
So that I can improve and fix my coding style faster.
```
## Problem

[JSCS](http://jscs.info/) has merged with [ESLint](http://eslint.org/), thus becoming deprecated. The project needs to migrate to ESLint. 
## Deliverables
- [ ] Enable ESLint and fix style accordingly.

### CODE REVIEW

**Comment by bogdan-stefan-hs** [comment]
@flavius-tirnacop-hs @valentin-radulescu-hs ESLint integration is ready.

I have disabled the `no-console` rule because `console` is used to log errors when the user enters invalid input in the editor, and also in some tests.
Are you ok with leaving it disabled, or do you have another suggestion?

Also, please review the code.

**Comment by valentin-radulescu-hs** [comment]
LGTM

Let's enable `no-console` if possible, we shouldn't rely on it in tests. Maybe disable it only for editor if you think it's useful there.

File: tests/.eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    /* testing objects */
+    'casper': false,
+    'flow': false,
+    'phantomCSS': false,
+    'document': false,
+    'sandbox': false,
+    'TestHelpers': false,
+    'sinon': false,
+    'window': false,
+    '_': false,
```
Comment (flavius-tirnacop-hs) [inline tests/.eslintrc.js]: I can't see why this should be global. If there aren't any hidden problems let's import lodash where needed.

**Comment by flavius-tirnacop-hs** [comment]
LGTM

**Pull request merged**